src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/functional/citizen/ClaimantResponseTest.java

```diff
@@ -59,6 +61,24 @@ public void shouldSaveClaimantResponseAcceptationReferToJudge() {
         assertThat(claimantResponse.getAmountPaid().orElse(ZERO)).isEqualTo(BigDecimal.TEN);
     }
 
+    @Test
+    public void shouldSaveClaimantResponseAcceptationReferToJudgeBecauseNoDisposableIncome() {
+        commonOperations.submitClaimantResponse(
+            ClaimantResponseAcceptation.builder().buildAcceptationWithCourtDeterminationWhenDisposableIncomeIsZero(),
+            claim.getExternalId(),
+            claimant
+        ).then()
+            .statusCode(HttpStatus.CREATED.value());
+
+        Claim claimWithClaimantResponse = commonOperations
+            .retrieveClaim(claim.getExternalId(), claimant.getAuthorisation());
+        ResponseAcceptation claimantResponse = (ResponseAcceptation) claimWithClaimantResponse.getClaimantResponse()
+            .orElseThrow(AssertionError::new);
+        Optional<CourtDetermination> courtDetermination = claimantResponse.getCourtDetermination();
+
+        assertThat(courtDetermination.isPresent());
```

Comment:
Could look like this using the 1.8 features - your discretion if it's easier to read or not
```
assertThat(claimWithClaimantResponse.getClaimantResponse()
        .filter(ResponseAcceptation.class::isInstance)
        .map(ResponseAcceptation.class::cast)
        .flatMap(ResponseAcceptation::getCourtDetermination)
        .isPresent());
```

Reply:
i like it but i will just try to write test for ResponseAcceptationContentProvider as that's the only file i am changing.